File: Builder/python-source/builders/HamburgerBuilder.py
from base.Burger import Burger
from builders.BurgerBuilder import BurgerBuilder


class HamburgerBuilder(BurgerBuilder):
    def __init__(self):
        super().__init__()
        self._contents = []

    def reset(self):
        self._contents = []

    def add_low_bread(self):
        self._contents.append('WhiteLoaf')

    def add_meat(self):
        self._contents.append('Farsh')

    def add_high_bread(self):
        self._contents.append('GrayLoaf')

    def add_vegetable(self):
        self._contents.append('Tomato')

    def get_burger(self) -> Burger:
        return Burger(self._contents)

File: JavaSource/src/main/java/ru/akhcheck/patterns/builder/builders/HamburgerBuilder.java
package ru.akhcheck.patterns.builder.builders;

import ru.akhcheck.patterns.builder.base.Burger;

import java.util.LinkedList;
import java.util.List;

public class HamburgerBuilder implements BurgerBuilder {
    private List<String> components = new LinkedList<>();

    @Override
    public BurgerBuilder reset() {
        components.clear();
        return this;
    }

    @Override
    public BurgerBuilder addLowBread() {
        components.add("WhiteLoaf");
        return this;
    }

    @Override
    public BurgerBuilder addMeat() {
        components.add("Farsh");
        return this;
    }

    @Override
    public BurgerBuilder addHighBread() {
        components.add("GrayLoaf");
        return this;
    }

    @Override
    public BurgerBuilder addVegetable() {
        components.add("Tomato");
        return this;
    }

    @Override
    public Burger getBurger() {
        return new Burger(components);
    }
}
